### PR TITLE
build: Check std::system for -[alert|block|wallet]notify

### PR DESCRIPTION
**Pull request by barton2526**
> Platforms such as iOs and Universal Windows Platform do not support launching a process through system().

Ref: https://github.com/bitcoin/bitcoin/pull/15457

Also includes fix `build: use #if HAVE_SYSTEM instead of defined(HAVE_SYSTEM)` from
https://github.com/bitcoin/bitcoin/pull/16344

Also includes `build: improve check for ::(w)system` from 
https://github.com/bitcoin/bitcoin/pull/22845

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1289,6 +1290,7 @@ bool SetBestChain(CTxDB& txdb, CBlock &blockNew, CBlockIndex* pindexNew) EXCLUSI
         pindexNew->nHeight,
         pindexNew->GetBlockTime(),
         blockNew.nBits);
```
Comment (div72) [inline src/main.cpp]: Including the uiInterface.NotifyBlocksChanged call in the ifdef breaks the GUI block counter for people who do not have std::system.

Comment (barton2526) [reply src/main.cpp]: should the `#endif` be placed on line 1286 instead?

Comment (div72) [reply src/main.cpp]: Yep.

**Review by div72 — APPROVED** [review]
ACK 82197b3f6a7ba66a2f1d061455a94137d13a0797.

**Pull request merged**